### ISSUE TITLE
`gh` cli now writes to stdout instead of stderr

### ISSUE THREAD

**Issue by domesticsimian** [PawelLipski/git-machete#985]
# Overview

As of version `2.31.0`, the `gh` cli now writes to stdout instead of stderr. See https://github.com/cli/cli/releases/tag/v2.31.0. It appears that git machete expects the token to be printed to stderr: https://github.com/VirtusLab/git-machete/blob/272708696059c13eea40744710a698b7f244f980/git_machete/github.py#L147

Note that there is also now a `gh auth token` command which is maybe a better way to get the token instead of parsing the output of `gh auth status`.

## Expected behavior/command output

Running `git machete github anno-prs`, I expect to see the PR annotations added.

## Actual behavior/command output

```
Checking for open GitHub PRs... GET https://api.github.com/repos/gro-intelligence/gro/pulls?per_page=100 request ended up in 404 response from GitHub. A valid GitHub API token is required.
Provide a GitHub API token with repo access via one of the: 
        1. GITHUB_TOKEN environment variable.
        2. Content of the ~/.github-token file.
        3. Current auth token from the gh GitHub CLI.
        4. Current auth token from the hub GitHub CLI.
 Visit https://github.com/settings/tokens to generate a new one.
```

## `git` and `git machete` versions

```
git --version
git version 2.34.1

git machete --version
git-machete version 3.17.4
```

## Additional diagnostics info

Relevant debug bits:

```
__get_token_from_env(cls=<class 'git_machete.github.GitHubToken'>): 1. Trying to authenticate via `GITHUB_TOKEN` environment variable...
__get_token_from_file_in_home_directory(cls=<class 'git_machete.github.GitHubToken'>, domain=github.com): 2. Trying to authenticate via `~/.github-token`...
__get_token_from_gh(cls=<class 'git_machete.github.GitHubToken'>, domain=github.com): 3. Trying to authenticate via `gh` GitHub CLI...
__get_token_from_hub(cls=<class 'git_machete.github.GitHubToken'>, domain=github.com): 4. Trying to authenticate via `hub` GitHub CLI...
Checking for open GitHub PRs... sync_annotations_to_github_prs(): Current GitHub user is <none>
```

**Comment by PawelLipski** [comment]
Whoops good spot 🐞  fixing now...

**Comment by PawelLipski** [comment]
To be released as v3.17.6 soon 🚢
